### PR TITLE
VideoCommon: Migrate over to fmt

### PR DESCRIPTION
**Pull request by lioncash**
Migrates off the printf-based formatting where applicable. This doesn't touch the frame dumping code, as that is being migrated by someone else in #9239

### CODE REVIEW

File: Source/Core/VideoCommon/BPStructs.cpp [hunk 1]
```diff
@@ -393,9 +394,9 @@ static void BPWritten(const BPCmd& bp)
     if (bp.changes & 12)
       PixelShaderManager::SetZTextureOpChanged();
 #if defined(_DEBUG) || defined(DEBUGFAST)
-    const char* pzop[] = {"DISABLE", "ADD", "REPLACE", "?"};
-    const char* pztype[] = {"Z8", "Z16", "Z24", "?"};
-    PRIM_LOG("ztex op=%s, type=%s", pzop[bpmem.ztex2.op], pztype[bpmem.ztex2.type]);
+    static constexpr const char* pzop[] = {"DISABLE", "ADD", "REPLACE", "?"};
+    static constexpr const char* pztype[] = {"Z8", "Z16", "Z24", "?"};
```
Comment (BhaaLseN) [inline Source/Core/VideoCommon/BPStructs.cpp:398]: Any particular reason you left those as `const char*[]` but changed the ones in `BPFunctions` to `std::string_view[]`?
Not poking you to change it (unless it makes sense to), I mostly wanna know if one has an advantage over the other, especially when they'd be considered equivalent.

File: Source/Core/VideoCommon/Fifo.cpp [hunk 2]
```diff
@@ -167,7 +167,8 @@ void SyncGPU(SyncGPUReason reason, bool may_move_read_ptr)
 
     // Opportunistically reset FIFOs so we don't wrap around.
     if (may_move_read_ptr && s_fifo_aux_write_ptr != s_fifo_aux_read_ptr)
-      PanicAlert("aux fifo not synced (%p, %p)", s_fifo_aux_write_ptr, s_fifo_aux_read_ptr);
+      PanicAlertFmt("Aux FIFO not synced ({}, {})", fmt::ptr(s_fifo_aux_write_ptr),
+                    fmt::ptr(s_fifo_aux_read_ptr));
```
Comment (BhaaLseN) [inline Source/Core/VideoCommon/Fifo.cpp:171]: Curlies, since this is now two lines inside the `if`?

**Review by BhaaLseN — APPROVED** [review]
Looks good in general, just a minor nit.

File: Source/Core/VideoCommon/BPStructs.cpp (hunk 1 above)
Comment (lioncash) [reply Source/Core/VideoCommon/BPStructs.cpp]: Just overlooked them. Changed these over as well.

File: Source/Core/VideoCommon/Fifo.cpp (hunk 2 above)
Comment (lioncash) [reply Source/Core/VideoCommon/Fifo.cpp]: Good catch

File: Source/Core/VideoCommon/VertexManagerBase.cpp [hunk 3]
```diff
@@ -381,19 +388,19 @@ void VertexManagerBase::Flush()
...
+  PRIM_LOG("frame{}:\n texgen={}, numchan={}, dualtex={}, ztex={}, cole={}, alpe={}, ze={}",
            g_ActiveConfig.iSaveTargetId, xfmem.numTexGen.numTexGens, xfmem.numChan.numColorChans,
            xfmem.dualTexTrans.enabled, bpmem.ztex2.op.Value(), bpmem.blendmode.colorupdate.Value(),
            bpmem.blendmode.alphaupdate.Value(), bpmem.zmode.updateenable.Value());
 
   for (u32 i = 0; i < xfmem.numChan.numColorChans; ++i)
   {
     LitChannel* ch = &xfmem.color[i];
-    PRIM_LOG("colchan%u: matsrc=%u, light=0x%x, ambsrc=%u, diffunc=%u, attfunc=%u", i,
+    PRIM_LOG("colchan%u: matsrc={}, light={:#x}, ambsrc={}, diffunc={}, attfunc={}", i,
```
Comment (leoetlino) [inline Source/Core/VideoCommon/VertexManagerBase.cpp:399]:
```suggestion
    PRIM_LOG("colchan{}: matsrc={}, light={:#x}, ambsrc={}, diffunc={}, attfunc={}", i,
```

File: Source/Core/VideoCommon/VertexManagerBase.cpp [hunk 4]
```diff
@@ -381,19 +388,19 @@ void VertexManagerBase::Flush()
...
              ch->matsource.Value(), ch->GetFullLightMask(), ch->ambsource.Value(),
              ch->diffusefunc.Value(), ch->attnfunc.Value());
     ch = &xfmem.alpha[i];
-    PRIM_LOG("alpchan%u: matsrc=%u, light=0x%x, ambsrc=%u, diffunc=%u, attfunc=%u", i,
+    PRIM_LOG("alpchan%u: matsrc={}, light={:#x}, ambsrc={}, diffunc={}, attfunc={}", i,
```
Comment (leoetlino) [inline Source/Core/VideoCommon/VertexManagerBase.cpp:403]:
```suggestion
    PRIM_LOG("alpchan{}: matsrc={}, light={:#x}, ambsrc={}, diffunc={}, attfunc={}", i,
```

**Review by leoetlino — CHANGES_REQUESTED** [review]
LGTM except in two places and for two PRIM_LOG calls in Source/Core/VideoCommon/VertexLoader_TextCoord.cpp that have been missed

**Comment by lioncash** [comment]
Addressed

**Review by leoetlino — APPROVED** [review]

**Pull request merged**